test/models/setting_test.rb

```diff
@@ -345,7 +345,7 @@ def test_boolean_true_returns_from_cache
     setting = Setting.where(:name => attrs[:name]).first || Setting.create(attrs)
     setting.value = ""
     assert !setting.save
-    assert_equal "URL must be valid and schema must be one of http and https", setting.errors[:value].first
+    assert_equal "can't be blank", setting.errors[:value].first
```

Comment:
Perhaps the `url_schema` validator should be moved before presence. As a user the current error is more useful than `can't be blank`.

Reply:
Would it really matter? because if the `url_schema` is not correct then it will through the appropriate error for the same.

Reply:
The difference is as a user you try an empty value and it tells you "can't be blank". Then try something and it tells you "URL must be valid and schema must be one of http and https`. In the current situation you all get best feedback in a single response.

Reply:
@ekohl what is your point? Rails validator always return all errors (in an array). Order does not matter, all errors are usually presented at once.

Reply:
Ah, I missed that it calls `.first`. I thought this was **the** error rather than an error.